primitives/src/merkle_tree/append_only.rs

```diff
@@ -31,17 +31,32 @@ use typenum::Unsigned;
 impl_merkle_tree_scheme!(MerkleTree);
 impl_forgetable_merkle_tree_scheme!(MerkleTree);
 
-impl<E, H, I, Arity, T> AppendableMerkleTreeScheme for MerkleTree<E, H, I, Arity, T>
+impl<E, H, I, Arity, T> MerkleTree<E, H, I, Arity, T>
 where
     E: Element,
     H: DigestAlgorithm<E, I, T>,
     I: Index + From<u64> + AddAssign + ToTraversalPath<Arity>,
```

Comment:
Whoa, TIL `ToTraversalPath` exists. it's all starting to fall into place. 😉 

I thought the plan is to force `Index = usize`. Also, do you not wish to remove `From<u64>`?

Reply:
Now we force `Index = u64`. [f7d75e4](https://github.com/EspressoSystems/jellyfish/pull/475/commits/f7d75e428bbd635e53f0eaf864ab54b63a35b9b6)